4_predict/src/plot_diagnostics.R

```diff
@@ -423,31 +423,61 @@ plot_timeseries <- function(pred_df, network_geometry, model_name, out_dir){
 
 
 #SHAP values
-plot_shap_global <- function(shap, model_name, out_dir, num_features = 40){
+plot_shap_global <- function(shap, model_name, out_dir, num_features = 40,
+                             dynamic_attrs_txt = NULL){
   #' 
   #' @description Creates SHAP global importance plot
   #'
   #' @param shap SHAP value results from compute_SHAP
   #' @param model_name character string describing the model. Will be added 
   #' to the end of the filename before the file extension, and also be the plot title.
   #' @param out_dir output directory
+  #' @param num_features number of features to plot
+  #' @param dynamic_attrs_txt Text file providing the dynamic attribute names as rows. 
+  #' When specified, makes a panel plot with static features
+  #' in one panel and dynamic attributes in another
   #'
-  #' @return Returns the paths to png files of SHAP dependence plots for each feature
+  #' @return Returns the paths to png files of SHAP global feature importance
   
+  #Plot for all attributes:
   fileout <- file.path(out_dir, 
-                       paste0('SHAP_global_', model_name, '.png'))
+                       paste0('SHAP_global_', model_name, '_vars', num_features, '.png'))
   
   p1 <- autoplot(shap, type = "importance", num_features = num_features) +
     ggtitle(model_name) + 
     theme(axis.text.y = element_text(size = 5))
   
   ggsave(filename = fileout, plot = p1, device = 'png')
   
+  if (!is.null(dynamic_attrs_txt)){
+    #Plot for panels of top static only and dynamic only attributes:
+    fileout <- c(fileout, 
+                 file.path(out_dir, 
+                           paste0('SHAP_global_panels_', model_name, '_vars', num_features, '.png'))
+                 )
+    
+    dynamic_attrs <- read_csv(dynamic_attrs_txt, col_names = FALSE,
+                             show_col_types = FALSE)
+    
+    x_lims <- c(0, ceiling(max(colMeans(abs(shap)))))
+    
+    p2 <- autoplot(shap[, -which(colnames(shap) %in% dynamic_attrs[[1]])], type = "importance", num_features = num_features) +
+      ggtitle(model_name, subtitle = 'Static Attributes') +
+      theme(axis.text.y = element_text(size = 5)) +
+      ylim(x_lims)
+    p3 <- autoplot(shap[, which(colnames(shap) %in% dynamic_attrs[[1]])], type = "importance", num_features = num_features) +
+      ggtitle(model_name, subtitle = 'Dynamic Attributes') +
+      theme(axis.text.y = element_text(size = 5)) +
+      ylim(x_lims)
+    
+    ggsave(filename = fileout[2], plot = p2+p3, device = 'png')
```

Comment:
This function always creates a plot with all features (combined static and dynamic), and that's where fileout[1] is used. It's not called as fileout[1] because fileout is a scaler at the time. I structured it this way because the vector of fileout is returned

Reply:
Ah, OK. I wasn't necessarily thinking of the targets implication and why we'd potentially need the vector. Thanks for clarifying!